File: Chagok/frontend/src/types/fact-summary.ts
/**
 * Fact Summary Types
 * 사건 전체 사실관계 요약 관련 타입 정의
 * Feature: 014-case-fact-summary
 */

/**
 * 사실관계 요약 응답 (GET /cases/{case_id}/fact-summary)
 */
export interface FactSummary {
  case_id: string;
  ai_summary: string;
  modified_summary: string | null;
  evidence_ids: string[];
  fault_types: string[];
  created_at: string;
  modified_at: string | null;
  modified_by: string | null;
  has_previous_version: boolean;
}

/**
 * 사실관계 생성 요청 (POST /cases/{case_id}/fact-summary/generate)
 */
export interface FactSummaryGenerateRequest {
  force_regenerate?: boolean;
}

/**
 * 사실관계 생성 응답
 */
export interface FactSummaryGenerateResponse {
  case_id: string;
  ai_summary: string;
  evidence_count: number;
  fault_types: string[];
  generated_at: string;
}

/**
 * 사실관계 수정 요청 (PATCH /cases/{case_id}/fact-summary)
 */
export interface FactSummaryUpdateRequest {
  modified_summary: string;
}

/**
 * 사실관계 수정 응답
 */
export interface FactSummaryUpdateResponse {
  case_id: string;
  modified_summary: string;
  modified_at: string;
  modified_by: string;
}

/**
 * 사실관계 패널 상태
 */
export type FactSummaryPanelState =
  | 'empty'        // 사실관계 없음
  | 'loading'      // 생성 중
  | 'generated'    // AI 생성 완료 (수정 전)
  | 'modified'     // 변호사 수정 완료
  | 'editing'      // 편집 모드
  | 'error';       // 에러 상태

/**
 * 사실관계 편집 상태
 */
export interface FactSummaryEditState {
  content: string;
  isDirty: boolean;
  isSaving: boolean;
}

File: Chagok/backend/app/schemas/fact_summary.py
"""
Fact Summary Schemas
014-case-fact-summary: Pydantic schemas for case fact summary API
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class FactSummaryBase(BaseModel):
    """사실관계 요약 기본 스키마"""
    ai_summary: str = Field(..., description="AI 생성 원본 사실관계")
    modified_summary: Optional[str] = Field(None, description="변호사 수정 사실관계")
    evidence_ids: List[str] = Field(default_factory=list, description="사용된 증거 ID 목록")
    fault_types: List[str] = Field(default_factory=list, description="추출된 유책사유")


class FactSummaryResponse(FactSummaryBase):
    """사실관계 조회 응답"""
    case_id: str
    created_at: datetime
    modified_at: Optional[datetime] = None
    modified_by: Optional[str] = None
    has_previous_version: bool = Field(False, description="이전 버전 존재 여부")


class FactSummaryGenerateRequest(BaseModel):
    """사실관계 생성 요청"""
    force_regenerate: bool = Field(False, description="기존 요약 무시하고 재생성")


class FactSummaryGenerateResponse(BaseModel):
    """사실관계 생성 응답"""
    case_id: str
    ai_summary: str
    evidence_count: int
    fault_types: List[str]
    generated_at: datetime


class FactSummaryUpdateRequest(BaseModel):
    """사실관계 수정 요청"""
    modified_summary: str = Field(..., min_length=1, max_length=15000)


class FactSummaryUpdateResponse(BaseModel):
    """사실관계 수정 응답"""
    case_id: str
    modified_summary: str
    modified_at: datetime
    modified_by: str
